library/std/src/net/tcp/tests.rs

```diff
@@ -46,6 +46,37 @@ fn connect_error() {
     }
 }
 
+#[test]
+fn connect_timeout_to_unreachable_address() {
```

Comment:
I tested, this unit test will cost at 2 minutes and 8 seconds on Linux platform, and cost 20 seconds on Windows.

Reply:
I think we only need one test here. We only need to focus on testing the problematic code path in one way. 2 minutes and 8 secs is a long time for a single unit test so I'd prefer if we just used `connect_timeout_error` test.

Reply:
Ok, I think 2 minutes and 8 seconds is a long time too.